AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -187,91 +192,186 @@ + (void)deleteDocumentWithPartition:(NSString *)partition
+  [self createWithPartition:partition
+                 documentId:documentId
+                   document:document
+               writeOptions:writeOptions
+          completionHandler:completionHandler];
+}
+
+- (void)readWithPartition:(NSString *)partition
+               documentId:(NSString *)documentId
+             documentType:(Class)documentType
+              readOptions:(MSReadOptions *)__unused readOptions
+        completionHandler:(MSDocumentWrapperCompletionHandler)completionHandler {
+  
+  // TODO consume writeOptions.
+  [MSTokenExchange
+   performDbTokenAsyncOperationWithHttpClient:(MSStorageIngestion *)self.ingestion
+   partition:partition
+   completionHandler:^(MSTokensResponse *_Nonnull tokenResponses, NSError *_Nonnull error) {
+     // If error getting token.
+     if (error || !tokenResponses) {
+       MSLogError([MSDataStore logTag], @"Can't get CosmosDb token:%@", [error description]);
+       completionHandler([[MSDocumentWrapper alloc] initWithError:error documentId:documentId]);
+       return;
+     }
+     
+     // Create http client.
+     MSCosmosDbIngestion *cosmosDbIngestion = [[MSCosmosDbIngestion alloc] init];
+     
+     // Call CosmosDb.
+     [MSCosmosDb
+      performCosmosDbAsyncOperationWithHttpClient:cosmosDbIngestion
+      tokenResult:tokenResponses.tokens[0]
+      documentId:documentId
+      httpMethod:@"GET"
+      body:nil
+      completionHandler:^(NSData *_Nonnull data, NSError *_Nonnull cosmosDbError) {
+        
+        // If not created.
+        if (!data || [MSDataSourceError errorCodeWithError:cosmosDbError] !=
+            kMSACDocumentSucceededErrorCode) {
+          MSLogError([MSDataStore logTag],
+                     @"Not able to read the document ID:%@ with error:%@",
+                     documentId, [cosmosDbError description]);
+          completionHandler([[MSDocumentWrapper alloc] initWithError:cosmosDbError
+                                                          documentId:documentId]);
+          return;
+        }
+        
+        // Deserialize.
+        NSError *deserializeError;
+        NSDictionary *json =
+        [NSJSONSerialization JSONObjectWithData:data
+                                        options:0
+                                          error:&deserializeError];
+        if (deserializeError) {
+          MSLogError([MSDataStore logTag], @"Error deserializing data:%@",
+                     [deserializeError description]);
+        }
+        MSLogDebug([MSDataStore logTag], @"Document json:%@", json);
+        
+        // Create document.
+        id<MSSerializableDocument> deserializedDocument = [(id<MSSerializableDocument>)[documentType alloc]
+                                                           initFromDictionary:(NSDictionary *)json[kMSDocumentKey]];
+        
+        // Create a document.
+        NSTimeInterval interval =
+        [(NSString *)json[kMSDocumentTimestampKey] doubleValue];
```

Comment:
formatting is wrong, please run the script

Reply:
fixed